File: 08-12-2024_Overlapping_Intervals.py
class Solution:
	def mergeOverlap(self, arr):
		#Code here
		arr.sort()  # Sort the array of intervals
        n = len(arr)
        res = []
        i = 0
        
        while i < n:
            temp = [0, 0]
            temp[0] = arr[i][0]
            temp[1] = arr[i][1]
            
            while i + 1 < n and temp[1] >= arr[i + 1][0]:
                temp[1] = max(temp[1], arr[i + 1][1])
                i += 1
            
            res.append(temp)
            i += 1
            
        return res

File: 08-12-2024_Overlapping_Intervals.java
class Solution {
    public List<int[]> mergeOverlap(int[][] arr) {
        // Code here 
        Arrays.sort(arr, (a, b) -> Integer.compare(a[0], b[0])); // Sort intervals by start time
        List<int[]> res = new ArrayList<>();
        int n = arr.length;
        int i = 0;
        
        while (i < n) {
            int start = arr[i][0];
            int end = arr[i][1];
            
            // Merge overlapping intervals
            while (i + 1 < n && end >= arr[i + 1][0]) {
                end = Math.max(end, arr[i + 1][1]);
                i++;
            }
            
            res.add(new int[]{start, end}); // Add merged interval as an int[] array
            i++;
        }
        
        return res;
    }
}